src/wp-includes/blocks/index.php

```diff
@@ -49,8 +49,34 @@ function register_core_block_style_handles() {
 		$files          = get_transient( $transient_name );
 		if ( ! $files ) {
 			$files = glob( wp_normalize_path( __DIR__ . '/**/**.css' ) );
+
+			/*
+			 * Make CSS files path relative before storing them in transient.
+			 *
+			 * Remove the `wp-includes` part to avoid issue when moving site (and the database) to another server :
+			 * /path/to/wp-includes/blocks/xxx/yyy.css => blocks/xxx/yyy.css
+			 */
+			$files = array_map(
+				static function ( $file ) use ( $includes_path ) {
+					return str_replace( $includes_path, '', $file );
+				},
+				$files
+			);
 			set_transient( $transient_name, $files );
 		}
+
+		/*
+		 * Rebuild the full path for all CSS file when loading them from transient.
+		 *
+		 * Add the `wp-includes` part back :
+		 * blocks/xxx/yyy.css => /path/to/wp-includes/blocks/xxx/yyy.css
+		 */
+		$files = array_map(
+			static function ( $file ) use ( $includes_path ) {
+				return $includes_path . $file;
+			},
+			$files
+		);
 	} else {
 		$files = glob( wp_normalize_path( __DIR__ . '/**/**.css' ) );
 	}
```

Comment:
I think we can optimize the logic here slightly for performance. We should consider that most commonly, sites will rely on the cache, because 99+% of sites do not use a development mode, or certainly not "core".

Instead of adding the `$includes_path` to every file from the cache, we should leave it out of the files and instead only add it in the `$register_style` closure below:
* Ensure `$files` never contains the `$includes_path` for its values.
* In `$register_style` closure, change the `$path` variable to not include `$includes_path` either.
* Only add the `$includes_path` in the final `$wp_styles->add_data()` call, similarly to how the `$includes_url` is added right above.

For the code here, this would mean something like as follows:
```suggestion
	} else {
		$files = glob( wp_normalize_path( __DIR__ . '/**/**.css' ) );
		$files = array_map(
				static function ( $file ) use ( $includes_path ) {
					return str_replace( $includes_path, '', $file );
				},
				$files
			);
	}
```

Reply:
@felixarntz in https://github.com/WordPress/wordpress-develop/pull/5052/commits/d93cd328b9e1733fc64ef99c272ab775ac367b53,
* I've changed `$files` to always use relative paths
* and updated the `in_array` to check against the `$style_path` variable.

I kept the `$path` variable since it's used in multiple places, (for `add_data` and for the `str_replace` for the RTL).